### PR TITLE
Add ChangeStepSize action

### PR DESCRIPTION
**Pull request by wthrowe**
This is the step size changing for LTS.  It does not cover changing the slab size or the step size for GTS evolutions (which will probably just be the slab size).

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Time/Actions/ChangeStepSize.hpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        Parallel::get<CacheTags::StepController>(cache);
+
+    const auto& current_step = db::get<Tags::TimeStep>(box);
+
+    double desired_step = std::numeric_limits<double>::infinity();
+    for (const auto& step_chooser : step_choosers) {
+      desired_step =
+          std::min(desired_step, step_chooser->desired_step(box, cache));
+    }
+    if (not current_step.is_positive()) {
```
Comment (kidder) [inline src/Time/Actions/ChangeStepSize.hpp:72]: comment (and verify doc) that StepChoosers always return positive values

**Review by kidder — CHANGES_REQUESTED** [review]
@nilsdeppe please look at

File: src/Time/Actions/ChangeStepSize.hpp (hunk 1 above)
Comment (wthrowe) [reply src/Time/Actions/ChangeStepSize.hpp:72]: StepChoosers already document this.  I've added a comment about the use here.

File: tests/Unit/Time/Actions/Test_ChangeStepSize.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Time.Actions.ChangeStepSize", "[Unit][Time][Actions]") {
+  const Slab slab(-5., -2.);
+  const double slab_length = slab.duration().value();
+  check(slab.start() + slab.duration() / 4, slab_length / 5.,
+        slab.duration() / 8);
+  check(slab.start() + slab.duration() / 4, slab_length, slab.duration() / 4);
+  check(slab.end() - slab.duration() / 4, slab_length / 5.,
```
Comment (kidder) [inline tests/Unit/Time/Actions/Test_ChangeStepSize.cpp]: I found this a bit confusing as you determine whether or not time is going forward or backward based on the sign of the expected step size passed into the check.  I would find a comment here from the call site helpful, or pass in the bool.

**Review by kidder — CHANGES_REQUESTED** [review]

File: tests/Unit/Time/Actions/Test_ChangeStepSize.cpp (hunk 2 above)
Comment (wthrowe) [reply tests/Unit/Time/Actions/Test_ChangeStepSize.cpp]: Went with a bool.

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Linux timeout during doc generation.

macOS timeout during cache saving.

**Pull request merged**